hazelcast/src/main/java/com/hazelcast/internal/diagnostics/DiagnosticsLogFile.java

```diff
@@ -107,36 +104,22 @@ private void renderStaticPlugins() throws IOException {
     }
 
     private void renderPlugin(DiagnosticsPlugin plugin) throws IOException {
-        logWriter.clean();
-        plugin.run(logWriter);
-        //bufferedWriter.append()
-        int desiredLength = charBuff.length;
-        int actualSize = logWriter.length();
-        while (desiredLength < actualSize) {
-            desiredLength *= 2;
-        }
+        logWriter.init(printWriter);
 
-        if (desiredLength != charBuff.length) {
-            charBuff = new char[desiredLength];
-        }
-
-        logWriter.copyInto(charBuff);
-        bufferedWriter.write(charBuff, 0, actualSize);
-        fileLength += actualSize;
+        plugin.run(logWriter);
     }
 
-    private BufferedWriter newWriter() throws FileNotFoundException {
+    private PrintWriter newWriter() throws FileNotFoundException {
         FileOutputStream fos = new FileOutputStream(file, true);
         CharsetEncoder encoder = Charset.forName("UTF-8").newEncoder();
-        return new BufferedWriter(new OutputStreamWriter(fos, encoder));
+        return new PrintWriter(new BufferedWriter(new OutputStreamWriter(fos, encoder), Short.MAX_VALUE));
```

Comment:
32 kb.

When the buffer is full it gets flushed, or unless it is explicitly flushed (e.g when the plugin has completed rendering its content)

Closing: yes.. that is how it works. So you can close the 'first' one and it automatically gets propagates; the same goes for the flush.